examples/dump_media/main.py

```diff
@@ -0,0 +1,201 @@
+    print("Download is complete")
+
+
+def log_exception(func):
+    def inner(*args):
+        try:
+            return func(*args)
+        except Exception as e:
+            args[0].logger.error(e)
+            raise
+
+    return inner
+
+
+class MediaDownloader:
+    def __init__(self, config, root_dir, log_to_file, dry_run):
+        self.public_api_client = PublicAPIClient(config)
+        self.log_to_file = log_to_file
+        self.dry_run = dry_run
+        self.root_dir = Path(root_dir)
+        self.root_dir.mkdir(parents=True, exist_ok=True)
+
+        if log_to_file:
+            logfile_path = f"{root_dir}/media_dump_{datetime.now().strftime('%Y-%m-%d_%H:%M:%S')}.log"
+            log_format = "%(asctime)s - %(levelname)s - %(message)s"
+            logging.basicConfig(
+                filename=logfile_path, level=logging.INFO, format=log_format
+            )
+        self.logger = logging.getLogger(__name__)
+
+    @log_exception
+    def get_all_media_ids_in_account(self):
+        response = self.search_media(1)
+        media_ids = [
+            media["id"]
+            for media in response.json()["media"]
+            if MediaDownloader.is_media_downloadable(media)
+        ]
+        last_page = response.json()["meta"]["last_page"]
+        total_count = response.json()["meta"]["total_count"]
+        self.logger.info(f"There are {total_count} videos on {last_page} page.")
+        if last_page > 1:
+            for page in range(2, last_page + 1):
+                response = self.search_media(page)
+                media_ids.extend(
+                    [
+                        media["id"]
+                        for media in response.json()["media"]
+                        if MediaDownloader.is_media_downloadable(media)
+                    ]
+                )
+        logger.info(f"Found media ids: {media_ids}")
+        return media_ids
+
+    def search_media(self, page):
+        self.logger.info(f"Getting {page}. page")
+        response = self.public_api_client.request(
+            "get", "media/search", params={"page": page}
+        )
+        self.raise_exception_if_not_ok(
+            response, f"Could not get all media on page {page}: {response.text}"
+        )
+        return response
+
+    def raise_exception_if_not_ok(self, response, error_message):
+        if response.status_code != 200:
+            self.logger.error(f"Abort: {response}")
+            raise Exception(error_message)
+
+    @staticmethod
+    def is_media_downloadable(media):
+        return media["transcoding_status"] == "transcoding_finished"
+
+    @log_exception
+    def extend_all_media_with_collection_and_captions(self, media_ids):
+        media_objects = []
+        for id in media_ids:
+            media_with_details = self.get_details_of_media(id)
+            captions = self.get_captions_of_media(id)
+            media_with_details.update(captions)
+            media_objects.append(media_with_details)
```

Comment:
how about:
```suggestion
            media_with_details = self.get_details_of_media(id).update(self.get_captions_of_media(id))
            media_objects.append(media_with_details)
```

Reply:
I wouldn’t think like that since the function name talks for itself `get_captions_of_media`.

Reply:
changed!